saleor/payment/gateways/stripe/webhooks.py

```diff
@@ -172,117 +153,134 @@ def _update_payment_with_new_transaction(
         raw_response=stripe_object.last_response,
         psp_reference=stripe_object.id,
     )
-    transaction = create_transaction(
-        payment,
-        kind=kind,
-        payment_information=None,  # type: ignore
-        action_required=False,
-        gateway_response=gateway_response,
-    )
-    gateway_postprocess(transaction, payment)
 
-    return transaction
+    transaction_id = stripe_object.id
+    if not payment.transactions.filter(
+        token=transaction_id,
+        kind=kind,
+        is_success=True,
+    ).exists():
+        transaction = create_transaction(
+            payment,
+            kind=kind,
+            payment_information=None,  # type: ignore
+            action_required=False,
+            gateway_response=gateway_response,
+        )
+        gateway_postprocess(transaction, payment)
+        return transaction
 
 
 def _process_payment_with_checkout(
     payment: Payment,
-    payment_intent: StripeObject,
-    kind: str,
-    amount: str,
-    currency: str,
+    manager: PluginsManager,
 ):
     checkout = _get_checkout(payment.id)
 
     if checkout and payment.can_create_order():
-        _finalize_checkout(checkout, payment, payment_intent, kind, amount, currency)
+        _finalize_checkout(checkout, payment, manager)
 
 
 def handle_authorized_payment_intent(
```

Comment:
Please double check that you've changed all definitions used for handling a notifications. Just want to be sure that we will not get an exception because we are trying to call function with incorrect definition

Reply:
This should be fine - there are both no gaps in coverage and no complaints from mypy